archunit/src/test/java/com/tngtech/archunit/core/PluginLoaderTest.java

```diff
@@ -1,41 +1,62 @@
 package com.tngtech.archunit.core;
 
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.HashSet;
 
 import com.tngtech.archunit.testutil.SystemPropertiesRule;
 import org.junit.Rule;
 import org.junit.Test;
 
+import static com.tngtech.archunit.core.PluginLoader.JavaVersion.JAVA_14;
 import static com.tngtech.archunit.core.PluginLoader.JavaVersion.JAVA_9;
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class PluginLoaderTest {
+    private static final Class<?> pluginTypeBeforeJava9 = HashSet.class;
+    private static final Class<?> pluginTypeBetweenJava9AndJava13 = ArrayList.class;
+    private static final Class<?> pluginTypeAfterJava13 = HashMap.class;
+
     @Rule
     public final SystemPropertiesRule systemPropertiesRule = new SystemPropertiesRule();
 
     @Test
     public void loads_correct_plugin_for_version() {
         System.setProperty("java.version", "1.7.0_55");
-        assertThat(loadsArrayListForJava9FallbackHashSet().load()).isInstanceOf(HashSet.class);
+        assertThat(loadsArrayListForJava9FallbackHashSet().load()).isInstanceOf(pluginTypeBeforeJava9);
 
         System.setProperty("java.version", "1.8.0_122");
-        assertThat(loadsArrayListForJava9FallbackHashSet().load()).isInstanceOf(HashSet.class);
+        assertThat(loadsArrayListForJava9FallbackHashSet().load()).isInstanceOf(pluginTypeBeforeJava9);
 
         System.setProperty("java.version", "9");
-        assertThat(loadsArrayListForJava9FallbackHashSet().load()).isInstanceOf(ArrayList.class);
+        assertThat(loadsArrayListForJava9FallbackHashSet().load()).isInstanceOf(pluginTypeBetweenJava9AndJava13);
 
         System.setProperty("java.version", "9.0.1");
-        assertThat(loadsArrayListForJava9FallbackHashSet().load()).isInstanceOf(ArrayList.class);
+        assertThat(loadsArrayListForJava9FallbackHashSet().load()).isInstanceOf(pluginTypeBetweenJava9AndJava13);
 
         System.setProperty("java.version", "11-ea");
-        assertThat(loadsArrayListForJava9FallbackHashSet().load()).isInstanceOf(ArrayList.class);
+        assertThat(loadsArrayListForJava9FallbackHashSet().load()).isInstanceOf(pluginTypeBetweenJava9AndJava13);
+
+        System.setProperty("java.version", "13");
+        assertThat(loadsArrayListForJava9FallbackHashSet().load()).isInstanceOf(pluginTypeBetweenJava9AndJava13);
+
+        System.setProperty("java.version", "14");
+        assertThat(loadsArrayListForJava9FallbackHashSet().load()).isInstanceOf(pluginTypeAfterJava13);
     }
 
     // PluginLoader memoizes the loaded plugin
     private PluginLoader<Object> loadsArrayListForJava9FallbackHashSet() {
```

Comment:
Now that this demo-`PluginLoader` returns one of the semantic variables, the name refering specifically to `ArrayList` and `HashSet` (but not `HashMap`) is a bit confusing.

How about naming this method `loadVersionDependentPluginType`?

Reply:
True, I think this should now also refer to the "meta names"...